lib/puppet/util/windows/eventlog.rb

```diff
@@ -31,7 +31,7 @@ class Puppet::Util::Windows::EventLog
   def initialize(source_name = 'Puppet')
     @eventlog_handle = RegisterEventSourceW(FFI::Pointer::NULL, wide_string(source_name))
     if @eventlog_handle == NULL_HANDLE
-      raise EventlogError.new("failed to open Windows eventlog")
+      raise EventLogError.new("RegisterEventSourceW failed to open Windows eventlog (exit code #{FFI.errno})", FFI.errno)
```

Comment:
Ha. Nice. So the mappings are just wrong on Windows..
```
However, operating systems typically report errors using plain integers. Module Errno is created dynamically to map these operating system errors to Ruby classes, with each error number generating its own subclass of SystemCallError
```
...from https://ruby-doc.org/core-2.2.0/Errno.html

Oh well. In that case there are some options, but they're not as clean. A couple include...write format error support. A really terrible option is since RuntimeException (nor any of the other exception class I glanced at) take an error code in the constructor, define EventLogError as an exception class that ignores the second parameter i.e., 
class EventLogError < RuntimeError
  def initialize(msg, code)
    super(msg)
  end
end

I'm leaning towards format error support but we're running out of time on this PR for 4.10